aptos-move/aptos-gas-schedule/src/gas_schedule/transaction.rs

```diff
@@ -249,12 +255,13 @@ impl TransactionGasParameters {
     pub fn calculate_intrinsic_gas(
         &self,
         transaction_size: NumBytes,
+        multiplier: NumBytes,
     ) -> impl GasExpression<VMGasParameters, Unit = InternalGasUnit> {
         let excess = transaction_size
             .checked_sub(self.large_transaction_cutoff)
             .unwrap_or_else(|| 0.into());
 
-        MIN_TRANSACTION_GAS_UNITS + INTRINSIC_GAS_PER_BYTE * excess
+        MIN_TRANSACTION_GAS_UNITS * multiplier + INTRINSIC_GAS_PER_BYTE * excess
```

Comment:
Yeah, another reason why it's cleaner to do a separate `keyless_charge`. If you do that you wouldn't have to hack the unit here.